### PR TITLE
fix: Allow dynamic properties on test struct

### PR DESCRIPTION
**Pull request by mitelg**
### 1. Why is this change necessary?

The tests are causing PHP deprecation notices

### 2. What does this change do, exactly?

- Allow dynamic properties on test struct
- remove assignment of `id` property

### 3. Describe each step to reproduce the issue or behaviour.

Look at any current PHPUnit pipeline

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [ ] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

### CODE REVIEW

File: tests/unit/Core/Framework/Struct/CollectionTest.php [hunk 1]
```diff
@@ -265,7 +265,7 @@ public function testFromAssociativeWithExpectedClass(): void
         $data = [
             'some-string',
             new \stdClass(),
-            ['id' => Uuid::randomHex(), 'versionId' => Uuid::randomHex()], // Entity class has no setId method
```
Comment (mitelg) [inline tests/unit/Core/Framework/Struct/CollectionTest.php:268]: I am not sure, if testing the `id` is important :thinking:

File: tests/unit/Core/Framework/Struct/Fixture/AssignTestStruct.php [hunk 2]
```diff
@@ -10,6 +10,7 @@
 /**
  * @internal
  */
+#[\AllowDynamicProperties]
```
Comment (mitelg) [inline tests/unit/Core/Framework/Struct/Fixture/AssignTestStruct.php:13]: only necessary, because `randomStringProperty` and `randomArrayProperty` are not declared as properties. I would assume this is intentional. if not, we should add them as normal properties

Comment (nfortier-shopware) [reply tests/unit/Core/Framework/Struct/Fixture/AssignTestStruct.php:13]: I also believe it was intentional

File: tests/unit/Core/Framework/Struct/CollectionTest.php (hunk 1 above)
Comment (nfortier-shopware) [reply tests/unit/Core/Framework/Struct/CollectionTest.php:268]: I don't think so, the focus is rather on versionId

**Review by nfortier-shopware — APPROVED** [review]

File: tests/unit/Core/Framework/Struct/Fixture/AssignTestStruct.php (hunk 2 above)
Comment (En0Ma1259) [reply tests/unit/Core/Framework/Struct/Fixture/AssignTestStruct.php:13]: Yes, it is intentional

File: tests/unit/Core/Framework/Struct/CollectionTest.php (hunk 1 above)
Comment (En0Ma1259) [reply tests/unit/Core/Framework/Struct/CollectionTest.php:268]: In short, this is some kind of testing an edge case.
The `versionId` is just a “random” value used to test whether the entity was created correctly. The main aspect of the test is adding only valid entities.

A normal `Entity` (e.g., `ProductEntity`) uses the `EntityIdTrait`. With this trait, specifying an `id` calls the `setId` method, which automatically sets `_uniqueIdentifier`. The `Entity` class however does not use `EntityIdTrait`, so `_uniqueIdentifier` is not set.
`_uniqueIdentifier` is required to add an entity to an `EntityCollection`.

Comment (mitelg) [reply tests/unit/Core/Framework/Struct/CollectionTest.php:268]: so, it can be left out or not? :sweat_smile:

Comment (En0Ma1259) [reply tests/unit/Core/Framework/Struct/CollectionTest.php:268]: I would keep it. It's an edge case behavior test.
It's different from e.g. this test case with only an `id`: https://github.com/shopware/shopware/blob/trunk/tests/unit/Core/Framework/Struct/AssignArrayTraitTest.php#L175

Comment (mitelg) [reply tests/unit/Core/Framework/Struct/CollectionTest.php:268]: but then, we will always get that PHP deprecation notice and it will not work with PHP 9. 

but I guess creating a new entity will blur again what should be tested?

Comment (En0Ma1259) [reply tests/unit/Core/Framework/Struct/CollectionTest.php:268]: Yeah, right. I guess it's fine to remove it

**Review by En0Ma1259 — APPROVED** [review]
